ic-asset/src/sync.rs

```diff
@@ -70,9 +79,13 @@ fn gather_asset_descriptors(dirs: &[&Path]) -> anyhow::Result<Vec<AssetDescripto
         let mut asset_descriptors_interim = vec![];
         for e in WalkDir::new(&dir)
             .into_iter()
-            .filter_entry(|entry| !filename_starts_with_dot(entry))
+            .filter_entry(|entry| {
+                doesnt_start_with_a_dot_or_explicitly_included(entry, &configuration)
+            })
```

Comment:
```suggestion
            .filter_entry(|entry|include_entry(entry, &configuration))
```

Reply:
change commited, but still would like to learn more about the idea

Reply:
I don't think removing `filter_entry` is a good idea, because there are cases where it gives a nice performance boost:
https://docs.rs/walkdir/latest/walkdir/struct.IntoIter.html#method.filter_entry
```
pub fn filter_entry(...) -> bool 
 
Yields only entries that satisfy the given predicate and skips descending into directories
that do not satisfy the given predicate.
The predicate is applied to all entries. If the predicate is true, iteration carries on as normal. 
If the predicate is false, the entry is ignored and if it is a directory, it is not descended into.
```

So if somebody would create a `.git` repo inside the assets dir, the directory would be skipped entirely thanks to `filter_entry`. 

Sadly, the `AssetSourceDirectoryConfiguration::config_map` would still get populated with all directories from `.git`, but that's because we went with `std::fs::read_dir(&dir)` instead of `walkdir`, therefore there is no way to take advantage of `filter_entry` there. We might want to change that in the future.

Reply:
e.g.
```rust
            .filter_entry(|entry| configuration.get_config(entry).is_included())
```

```rust
impl AssetSourceDirectoryConfiguration {
  fn get_config -> ConfigFileResult<AssetConfig> {...}
}

enum ConfigFileResult {
  NotFound,
  Excluded,
  Included<AssetConfig>
}

impl ConfigFileResult {
  is_included(&self) -> bool { 
    self == Self::Included 
} //pseudo
}
```
?

Reply:
done in https://github.com/dfinity/agent-rs/pull/361/commits/ffc5abe70058880f9f92c5ebfb3f215944e834ed

turned out great, thanks for sharing your idea!